weld/llvm.rs

```diff
@@ -901,7 +903,22 @@ impl LlvmGenerator {
                     self.prelude_code.add(&name_replaced);
                     self.prelude_code.add("\n");
 
-                    // Supports vectorization, so splice in the vector extensions.
+                    // Generate comparison functions, using a special template for vec[i8].
+                    if *elem.as_ref() == Scalar(ScalarKind::I8) {
```

Comment:
Yeah, that's true, that's one reason I separated "eq" from "cmp". Even then though, it's technically incorrect for floats because NaN != NaN; we need to see how we want to handle that (but I believe our built in comparisons do the right thing there).